mode/sparql/sparql.js

```diff
@@ -33,6 +33,15 @@ CodeMirror.defineMode("sparql", function(config) {
                              "true", "false", "with",
                              "data", "copy", "to", "move", "add", "create", "drop", "clear", "load", "into"]);
   var operatorChars = /[*+\-<>=&|\^\/!\?]/;
+  var PN_CHARS_BASE =
+    "[A-Z]|[a-z]|[\\u{00C0}-\\u{00D6}]|[\\u{00D8}-\\u{00F6}]|[\\u{00F8}-\\u{02FF}]|" +
+    "[\\u{0370}-\\u{037D}]|[\\u{037F}-\\u{1FFF}]|[\\u{200C}-\\u{200D}]|[\\u{2070}-\\u{218F}]|" +
+    "[\\u{2C00}-\\u{2FEF}]|[\\u{3001}-\\u{D7FF}]|[\\u{F900}-\\u{FDCF}]|[\\u{FDF0}-\\u{FFFD}]|[\\u{10000}-\\u{EFFFF}]";
```

Comment:
Could you simplify these a bit by not repeating the `[]` characters? (`[A-Za-Z]` is equivalent to `[A-Z]|[a-z]`).

Reply:
I have eliminated many of the square brackets: I have grouped the characters named PN_CHARS_BASE, PN_CHARS_U and PN_CHARS in the specification EBNF together as character classes, so there is still some use of "|".